### PR TITLE
Allow settings to be updated and read.

### PR DESCRIPTION
**Pull request by spacedmonkey**
Working on #48885. Decided to go a different direction than the ticket name. Use the existing endpoint but add some new callbacks. 

To use this PR, here is the example I use to test. 

```
add_action('init', function(){
	register_setting(
		'general',
		'wibble',
		array(
			'show_in_rest' => array(
				'name' => 'wibble',
				'edit_permission_callback' => '__return_true',
				'read_permission_callback' => '__return_true',
			),
			'type'         => 'string',
			'description'  => __( 'Site title.' ),
		)
	);
});
```
This make a setting called `wibble`, that will now be public to read and edit. You never do this, but it gives you an idea of how the code works. 

Trac ticket: https://core.trac.wordpress.org/ticket/48885
---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Comment by spacedmonkey** [comment]
@TimothyBJacobs This PR is a work in progress, but gives you an idea of what i am thinking. If you are happy with the way I did this, I will add tests.

**Comment by TimothyBJacobs** [comment]
This path looks good to me!

We should make sure that if the user passed an invalid callback the option isn't disclosed.

File: src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php [hunk 1]
```diff
@@ -66,9 +66,49 @@ public function register_routes() {
...
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to update the item, WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		$options = $this->get_registered_options();
+
+		foreach ( $options as $name => $args ) {
+			if ( isset( $request[ $name ] ) ) {
+				$test = call_user_func_array( $args['edit_permission_callback'], array( $name ) );
+				if ( ! $test ) {
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php]:
```suggestion
				$has_permissions = call_user_func_array( $args['edit_permission_callback'], array( $name ) );
				if ( ! $has_permissions ) {
```

File: src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php [hunk 2]
```diff
@@ -66,9 +66,49 @@ public function register_routes() {
...
+						array(
+							'key'    => $name,
+							'status' => rest_authorization_required_code(),
+						)
+					);
+				}
+			}
+		}
+
+		return true;
```
Comment (spacedmonkey) [inline src/wp-includes/rest-api/endpoints/class-wp-rest-settings-controller.php]:
```suggestion
		return current_user_can( 'manage_options' );
```

**Comment by adamziel** [comment]
The tests are all failing unfortunately :(

**Pull request closed without merging**